uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -488,11 +481,16 @@ define([
     function setCalendarProps(contactID, calendar) {
       var dateObj,
         leaveRequest,
-        dates = Object.keys(calendar.days);
+        dates = Object.keys(calendar.days),
+        indexedByMonth = angular.copy(vm.months.map(function () {
+          return [];
+        }));
 
       dates.forEach(function (date) {
         dateObj = calendar.days[date];
```

Comment:
This could be streamlined and made it simpler

```js
function setCalendarProps(contactID, calendar) {
  var indexedByMonth = vm.months.map(function () { return []; });

  _.each(calendar.days, function (dateObj) {
    var leaveRequest;

    // ...
  });

  return indexedByMonth;
}
```